tests/test_doctestplus.py

```diff
@@ -775,3 +775,137 @@ def test_doctest_skip(testdir):
         """
     )
     testdir.inline_run(p, '--doctest-plus', '--doctest-rst').assertoutcome(skipped=1)
+
+
+def test_remote_data_on(testdir):
+    testdir.makeini(
+        """
+        [pytest]
+        doctestplus = enabled
+    """)
+
+    p = testdir.makefile(
+        '.rst',
+        """
+        # This test should pass
+        .. doctest-remote-data::
+            >>> 1 + 1
+            2
+        """
+    )
+    testdir.inline_run(p, '--doctest-plus', '--doctest-rst',
+                       '--remote-data').assertoutcome(passed=1)
+
+
+def test_remote_data_off(testdir):
+    testdir.makeini(
+        """
+        [pytest]
+        doctestplus = enabled
+    """)
+
+    p = testdir.makefile(
+        '.rst',
+        """
+        # This test should fail, but we skip it, thus the test should pass.
+        .. doctest-remote-data::
+            # Now do real remote access:
+            >>> from contextlib import closing
+            >>> from urllib.request import urlopen
+            >>> with closing(urlopen('https://www.astropy.org')) as remote:
+            ...     remote.read()    # doctest: +IGNORE_OUTPUT
+            #
+            # And do some failing to check whether the whole block is indeed skipped
+            >>> 1 + 1
+            3
+            """
+    )
+    testdir.inline_run(p, '--doctest-plus', '--doctest-rst').assertoutcome(skipped=1)
+
+
+def test_remote_data_float_cmp(testdir):
+    testdir.makeini(
+        """
+        [pytest]
+        doctestplus = enabled
+    """)
+
+    p = testdir.makefile(
+        '.rst',
+        """
+        #This test is skipped when remote data is not requested, and should
+        # fail when remote data is requested
+        .. doctest-remote-data::
+            >>> x = 1/3.
+            >>> x    # doctest: +FLOAT_CMP
+            0.333333
+            >>> 1 + 1
+            3
+            """
+    )
+    testdir.inline_run(p, '--doctest-plus', '--doctest-rst').assertoutcome(skipped=1)
+
+
+def test_remote_data_ellipsis(testdir):
+    testdir.makeini(
+        """
+        [pytest]
+        doctest_optionflags = ELLIPSIS
```

Comment:
What is the purpose of the extra `doctest_optionflags = ELLIPSIS` in this test?

Reply:
I was trying to test if the ELLIPSIS option works well with the `.. doctest-remote-data::` directive, but I think I misunderstood it's function. The correct option here should be `NORMALIZE_WHITESPACE `

Reply:
@pllim - we run into some cases where it seemed some of the default flags don't work nicely together, so as mentioned above the purpose was to test some of them. 
Maybe the test is superfluous, then we can remove it later.